packages/server/test/utils/xdccDatabase.test.ts

```diff
@@ -130,6 +132,49 @@ invalid line
 
 			expect(mockPrepare).toHaveBeenCalled();
 		});
+
+		it("should filter out excluded channels", async () => {
+			config.excludedChannels = new Set(["#a-b-c"]);
+
+			mockFetch.mockResolvedValue({
+				text: () => Promise.resolve(`#1   Bot1 100M TestFile.txt`),
+			} as Response);
+
+			await create(parsedXdccDatabase);
+
+			expect(mockFetch).toHaveBeenCalledTimes(1);
+			expect(mockFetch).toHaveBeenCalledWith("http://www.pierpaolo.org/scripts.php");
+
+			config.excludedChannels = new Set();
+		});
+
+		it("should filter out multiple excluded channels", async () => {
+			config.excludedChannels = new Set(["#a-b-c", "#Pierpaolo"]);
+
+			mockFetch.mockResolvedValue({
+				text: () => Promise.resolve(`#1   Bot1 100M TestFile.txt`),
+			} as Response);
+
+			await create(parsedXdccDatabase);
+
+			expect(mockFetch).not.toHaveBeenCalled();
+
+			config.excludedChannels = new Set();
+		});
+
+		it("should not filter when excludedChannels is empty", async () => {
+			config.excludedChannels = new Set();
+
+			mockFetch.mockResolvedValue({
+				text: () => Promise.resolve(`#1   Bot1 100M TestFile.txt`),
+			} as Response);
+
+			await create(parsedXdccDatabase);
+
+			expect(mockFetch).toHaveBeenCalledTimes(2);
+
+			config.excludedChannels = new Set();
```

Comment:
Can we move this into an after each? It is repeated multiple times in these tests.